### ISSUE TITLE
Netbox Plugin

### ISSUE THREAD

**Issue by ryanmerolle** [respawner/peering-manager#255]
### Proposed Functionality

As of 2.8 NetBox released a plugins feature to extend the functionality of Netbox.  It would be amazing if we could display peering manager information in NetBox.  Features could include:

- Build a separate netbox plugin to display peering-manager summaries for routers found in peering-manager.
- Aggregate a view of all peering counts and stats per site based on the routers associated to said site.

### Use Case

Tighter netbox and peering-manager integration.

### External Dependencies

This is likely  another project into itself.

**Comment by respawner** [comment]
I agree that it is likely another project. Maybe it's time to create a new organization on GitHub to group Peering Manager projects such as the initial project itself, the Ansible role and the potentially new NetBox plugin.

**Comment by ryanmerolle** [comment]
That’s a good point about the org.

**Comment by gmazoyer** [comment]
Reviewing what a NetBox plugin can do, we should probably have a go at this. NetBox could therefore send data to Peering Manager and even retrieve them for display. IMO the plugin should be as less intrusive as possible, we do not want to invade NetBox space too much.

**Comment by tweippert** [comment]
Has someone started here already? 

I'm currently looking on some other plugin ideas on netbox, and maybe i can volunteer in the project? 
Will try to get some of @ryanmerolle ideas in a first initial plugin for getting some informations from peering manager belongs to routers also present in peering manager.

For the site informations, i think it is necessary to summarize all routers within a site (and also in netbox)  to aggregate the informations, as peering manager has no direct site configuration (think IXP not necessarily mean a site in netbox).

**Comment by gmazoyer** [comment]
There is an ongoing discussion #498, feel free to share your thoughts over there. We are gathering ideas before actually rewriting some code.
